app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeScreen.kt

```diff
@@ -198,10 +214,10 @@ private fun MyCoins(
                 contentPadding = PaddingValues(horizontal = Dp16),
                 horizontalArrangement = Arrangement.spacedBy(Dp16)
             ) {
-                items(coins) { coin ->
+                itemsIndexed(coins) { index, coin ->
                     CoinItem(
                         coinItem = coin,
-                        onMyCoinsItemClick = onMyCoinsItemClick
+                        onItemClick = { onMyCoinsItemClick.invoke(index) }
```

Comment:
Should we pass `coin` model instead of its index?

```suggestion
                        onItemClick = { onMyCoinsItemClick.invoke(coin) }
```